src/java/org/apache/cassandra/service/MigrationManager.java

```diff
@@ -188,6 +207,24 @@ public static boolean shouldPullSchemaFrom(InetAddress endpoint)
                 && !Gossiper.instance.isGossipOnlyMember(endpoint);
     }
 
+    public static boolean shouldPullSchemaFrom(InetAddress endpoint, UUID theirVersion)
+    {
+        /*
+         * Don't request schema from nodes with a differnt or unknonw major version (may have incompatible schema)
+         * Don't request schema from fat clients
+         * Don't request schema from bootstrapping nodes (?)
+         * Don't request schema if we have scheduled a pull request for that schema version
+         */
+        boolean isOtherSchemaNonEmpty = !Schema.emptyVersion.equals(theirVersion);
+        boolean noScheduledRequests = !scheduledSchemaPulls.contains(theirVersion);
+        logger.debug("Evaluating schema pull criteria: other schema empty {}, no scheduled requests {}", isOtherSchemaNonEmpty, noScheduledRequests);
```

Comment:
This was added for testing purposes... reminder to remove before merging. Also, got the log param message wrong 🤦‍♂️